### ISSUE TITLE
Problem with Azure OpenAI chatbot on your own data and Teams Toolkit

### ISSUE THREAD

**Issue by AlexS640** [OfficeDev/TeamsFx#12904]
**Describe the bug**
I am using Azure Open AI Studio to generate an AI chatbot 'on your own data' and trying to deploy it as a Teams app (see [link](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/use-your-data?tabs=ai-search%2Cteams#deploy-to-a-copilot-preview-teams-app-preview-or-web-app)). When I run the chatbot locally with the Teams Toolkit in Visual Studio Code (with 'Debug in Test Tool' configuration), it works fine. However, after provisioning, deploying and publishing it to our Azure Cloud tenant, the respective Azure Bot does not work. When I go to 'Settings -> Test in Web Chat' in the respective Azure resource and input a message, nothing happens. When looking at 'Settings -> Channels -> Web Chat', I can see a number of issues: 
![image](https://github.com/user-attachments/assets/5bdedac4-0c31-49ea-ae58-4ac48c6083b3)
Also when I manually import the resulting .zip file in my local 'appPackage/build' directory into Teams as an app, I obtain no answers. I think, it could be the same problem there.

**To Reproduce**
1. Follow the steps specified in [link](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/use-your-data?tabs=ai-search%2Cteams#deploy-to-a-copilot-preview-teams-app-preview-or-web-app)
to deploy your 'Azure OpenAI chatbot on your own data' as a Teams app.
2. Go to the Azure portal, where the Azure bot is deployed, and select the respective resource.
3. Go to 'Test in Web Chat' and input a message.
4. Error: No answer is generated.
5. Go to 'Channels' and look for issues in the 'Web Chat' channel.

**Expected behavior**
I expect an answer, when using the 'Test in Web Chat' functionality.

**Screenshots**
![image](https://github.com/user-attachments/assets/97a32b81-7fb8-415a-b3cf-215d6cc64ae5)

**VS Code Extension Information (please complete the following information):**
 - OS: Windows 
 - Version 5.10.1

**CLI Information (please complete the following information):**
 - OS: Windows 10
 - Version 22H2

**Comment by Prasad-MSFT** [comment]
Hi @AlexS640, thanks for raising your issue. Does this issue happen only with 'Web Chat', or does it happen in 'Teams chat' as well? The error message "HTTP status code Unauthorized" typically indicates that there is a problem with the bot's credentials or configuration. Could you please double-check that the bot is configured with the correct values for the app ID and password? These values should match the ones specified in the Azure Bot Service configuration. Sometimes, the issue might be due to missing or incorrect values in the bot's configuration settings.

**Comment by AlexS640** [comment]
Hi @Prasad-MSFT, thank you for your reply. It also happens in the Teams chat. The bot type is 'User-assigned Managed Identity' which was automatically configured during the Azure cloud deployment step, so I think there is no app password. I also followed the guidlines to give the respective managed identity the 'Cognitive Service OpenAI user' role on the resource group hosting the underlying Azure OpenAI service.

**Comment by Prasad-MSFT** [comment]
Hi @AlexS640, please share repro video of your issue along with bot id and timestamp of failure, so that we can raise a bug for this issue.

**Comment by AlexS640** [comment]
Hi @Prasad-MSFT, unfortunately I cannot share a repro video here, because it could potentially contain confidential data. Is there another possibility?

**Comment by Prasad-MSFT** [comment]
@AlexS640 You can share it via microsoftteamsdev@microsoft.com

**Comment by AlexS640** [comment]
Sorry for my late reply and thank you for the information @Prasad-MSFT. The Azure tenant, where I am deploying the underlying resources (Bot, App Service, ...) is different from the one where the authentication for Teams is managed. Could that be the reason for my problem?

**Comment by blackchoey** [comment]
@AlexS640 Some checks in the generated project caused your issue
1. The first check on authorization header will cause Test in Web not work
2. The second check on the conversation tenant id will cause no response in your Teams, as you mentioned your Teams tenant and Azure tenant are different.


Below is the code (index.ts) I obtained from Azure AI Foundry. Please let me know if you got different code.
```
    if (!req.headers.authorization && process.env.RUNNING_ON_AZURE === '1') {
        return res.send(401, 'unauthorized');
    }

    // Route received a request to adapter for processing
    await (app.adapter as TeamsAdapter).process(req, res, async (context: TurnContext) => {
        if (
            context.activity.conversation.tenantId !== process.env.BOT_TENANT_ID &&
            process.env.RUNNING_ON_AZURE === '1'
        ) {
            return res.send(401, 'invalid tenant');
        }

        // Dispatch to application for routing
        await app.run(context);
    });
```

**Comment by AlexS640** [comment]
@blackchoey Thank you for the information. Yes, I have the same code parts in the index.ts of my Teams Toolkit App.

**Comment by blackchoey** [comment]
I believe remove these logic should solve your issues. Please let us know if you meet other issues.

**Comment by AlexS640** [comment]
@blackchoey Good point, thank you. I am going to try that in the near future and tell you about my experience.
